Tasks/AzureCLIV2/azureclitask.ts

```diff
@@ -0,0 +1,370 @@
+                fs.writeFileSync(cliPassword, certificateContent);
+                this.cliPasswordPath = cliPassword;
+            }
+            else {
+                tl.debug('key based endpoint');
+                cliPassword = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
+                this.servicePrincipalKey = cliPassword;
+            }
+
+            //login using svn
+            Util.throwIfError(tl.execSync("az", "login --service-principal -u \"" + servicePrincipalId + "\" -p \"" + cliPassword + "\" --tenant \"" + tenantId + "\""), tl.loc("LoginFailed"));
+        }
+        else if(authScheme.toLowerCase() == "managedserviceidentity") {
+            //login using msi
+            Util.throwIfError(tl.execSync("az", "login --identity"), tl.loc("MSILoginFailed"));
+        }
+        else{
+            throw tl.loc('AuthSchemeNotSupported', authScheme);
+        }
+
+        this.isLoggedIn = true;
+        //set the subscription imported to the current subscription
+        Util.throwIfError(tl.execSync("az", "account set --subscription \"" + subscriptionID + "\""), tl.loc("ErrorInSettingUpSubscription"));
+    }
+
+    private static setConfigDirectory(): void {
+        if (tl.getBoolInput("useGlobalConfig")) {
+            return;
+        }
+
+        if (!!tl.getVariable('Agent.TempDirectory')) {
+            var azCliConfigPath = path.join(tl.getVariable('Agent.TempDirectory'), ".azclitask");
+            console.log(tl.loc('SettingAzureConfigDir', azCliConfigPath));
+            process.env['AZURE_CONFIG_DIR'] = azCliConfigPath;
+        } else {
+            console.warn(tl.loc('GlobalCliConfigAgentVersionWarning'));
+        }
+    }
+
+    private static setAzureCloudBasedOnServiceEndpoint(): void {
+        var connectedService: string = tl.getInput("connectedServiceNameARM", true);
+        var environment = tl.getEndpointDataParameter(connectedService, 'environment', true);
+        if(!!environment) {
+            console.log(tl.loc('SettingAzureCloud', environment));
+            Util.throwIfError(tl.execSync("az", "cloud set -n " + environment));
+        }
+    }
+
+    private static logoutAzure() {
+        try {
+            tl.execSync("az", " account clear");
+        }
+        catch (err) {
+            // task should not fail if logout doesn`t occur
+            tl.warning(tl.loc("FailedToLogout"));
+        }
+    }
+}
+
+class ScriptTypeFactory {
+    public static getSriptType(): ScriptType {
+        let scriptType: string = tl.getInput("scriptType");
+        let scriptLocation: string = tl.getInput("scriptLocation");
+        let scriptArguments: string = tl.getInput("scriptArguments", false);
+        switch(scriptType){
+            case 'ps': 
+                return new PowerShell5(scriptLocation, scriptArguments);
+            case 'pscore': 
+                return new PowerShell6(scriptLocation, scriptArguments);
```

Comment:
Call it `PowerShellCore`. it might become PowerShell in 7 but still we can internally use Core word